### PR TITLE
 [OSDOCS-12383]Add PSC option to CCS infrastructure models in OSD GCP docs

### PR DESCRIPTION
**Pull request by mletalie**
Version(s):

4.17+
Issue:

https://issues.redhat.com/browse/OSDOCS-12383
Link to docs preview:

https://84143--ocpdocs-pr.netlify.app/openshift-dedicated/latest/osd_install_access_delete_cluster/creating-a-gcp-cluster.html#osd-create-cluster-rhm-gcp-account_osd-creating-a-cluster-on-gcp
https://84143--ocpdocs-pr.netlify.app/openshift-dedicated/latest/osd_install_access_delete_cluster/creating-a-gcp-cluster.html#osd-create-cluster-gcp-account_osd-creating-a-cluster-on-gcp
https://84143--ocpdocs-pr.netlify.app/openshift-dedicated/latest/osd_install_access_delete_cluster/creating-a-gcp-cluster.html#osd-create-gcp-cluster-ccs_osd-creating-a-cluster-on-gcp

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/osd-create-cluster-rhm-gcp-account.adoc [hunk 1]
```diff
@@ -90,22 +90,37 @@ After your cluster is created, you can change the number of compute nodes, but y
 
 . Click *Next*.
 
-. In the *Cluster privacy* dialog, select *Public* or *Private* to use either public or private API endpoints and application routes for your cluster.
+. In the *Cluster privacy* dialog, select *Public* or *Private* to use either public or private API endpoints and application routes for your cluster. If you select *Private*, the optional *Use Private Service Connect* feature is automatically enabled.
+If you do not want to use Private Service Connect (PSC), disable by clicking the *Use Private Service Connect* checkbox.
```
Comment (AedinC) [inline modules/osd-create-cluster-rhm-gcp-account.adoc:94]:
```suggestion
You can disable PSC by clicking the *Use Private Service Connect* checkbox.
```

File: modules/osd-create-cluster-rhm-gcp-account.adoc [hunk 2]
```diff
@@ -90,22 +90,37 @@ After your cluster is created, you can change the number of compute nodes, but y
 
 . Click *Next*.
 
-. In the *Cluster privacy* dialog, select *Public* or *Private* to use either public or private API endpoints and application routes for your cluster.
+. In the *Cluster privacy* dialog, select *Public* or *Private* to use either public or private API endpoints and application routes for your cluster. If you select *Private*, the optional *Use Private Service Connect* feature is automatically enabled.
```
Comment (AedinC) [inline modules/osd-create-cluster-rhm-gcp-account.adoc:93]:
```suggestion
. In the *Cluster privacy* dialog, select *Public* or *Private* to use either public or private API endpoints and application routes for your cluster. If you select *Private*,  *Use Private Service Connect* is selected by default. Private Service Connect (PSC) is Google Cloud’s security-enhanced networking feature.
```

File: modules/osd-create-cluster-rhm-gcp-account.adoc [hunk 3]
```diff
@@ -90,22 +90,37 @@ After your cluster is created, you can change the number of compute nodes, but y
...
+If you do not want to use Private Service Connect (PSC), disable by clicking the *Use Private Service Connect* checkbox.
++
+[NOTE]
+====
+Red Hat recommends Private Service Connect (PSC) for network connectivity.
```
Comment (AedinC) [inline modules/osd-create-cluster-rhm-gcp-account.adoc:98]:
```suggestion
Red Hat recommends using PSC for more secure network connectivity.
```
Brain not fully braining here again but don't feel it's strong enough in it's current iteration. An alternative suggestion would be: PSC is Red Hat's recommended form of network connectivity.

File: modules/osd-create-cluster-rhm-gcp-account.adoc [hunk 4]
```diff
@@ -90,22 +90,37 @@ After your cluster is created, you can change the number of compute nodes, but y
...
+====
+If you selected *Use Private Service Connect* as part of your network configuration, you must select *Install into an existing VPC*.
+====
++
 .. If you are installing into an existing VPC and you want to enable an HTTP or HTTPS proxy for your cluster, select *Configure a cluster-wide proxy*.
-
++
+[IMPORTANT]
+====
+In order to configure a cluster-wide proxy for your cluster, you must first create the Cloud network address translation (NAT) and a Cloud router. See the "Additional resources" section for more information.
```
Comment (AedinC) [inline modules/osd-create-cluster-rhm-gcp-account.adoc:114]:
```suggestion
In order to configure a cluster-wide proxy for your cluster, you must first create the Cloud network address translation (NAT) and a Cloud router. See the _Additional resources_ section for more information.
```

File: modules/osd-create-cluster-rhm-gcp-account.adoc [hunk 5]
```diff
@@ -90,22 +90,37 @@ After your cluster is created, you can change the number of compute nodes, but y
...
 +
-
-. Optional: To install the cluster into a GCP Shared VPC:
+. Optional: To install the cluster into a GCP shared VPC:
 +
 [IMPORTANT]
 ====
-
-To install a cluster into a Shared VPC, you must use {product-title} version 4.13.15 or above. Additionally, the VPC owner of the host project must enable a project as a host project in their Google Cloud console. For more information, see link:https://cloud.google.com/vpc/docs/provisioning-shared-vpc#set-up-shared-vpc[Enable a host project].
+To install a cluster into a GCP shared VPC, you must use {product-title} version 4.13.15 or above. Additionally, the VPC owner of the host project must enable a project as a host project in their Google Cloud console. For more information, see link:https://cloud.google.com/vpc/docs/provisioning-shared-vpc#set-up-shared-vpc[Enable a host project].
```
Comment (AedinC) [inline modules/osd-create-cluster-rhm-gcp-account.adoc:123]:
```suggestion
To install a cluster into a GCP shared VPC, you must use {product-title} version 4.13.15 or later. Additionally, the VPC owner of the host project must enable a project as a host project in their Google Cloud console. For more information, see link:https://cloud.google.com/vpc/docs/provisioning-shared-vpc#set-up-shared-vpc[Enable a host project].
```

**Review by AedinC — COMMENTED** [review]
Done- looks great. Just a couple of very minor suggestions.

File: modules/osd-create-cluster-rhm-gcp-account.adoc (hunk 3 above)
Comment (mletalie) [reply modules/osd-create-cluster-rhm-gcp-account.adoc:98]: Brain not Braining! :)

**Comment by svmrh** [comment]
@mletalie , @AedinC : 

Can we make this note consistent with the tooltip added in the UI? 
_"Red Hat recommends using Private Service Connect when deploying a private OpenShift Dedicated cluster on Google Cloud. Private Service Connect ensures there is a secured, private connectivity between Red Hat infrastructure, Site Reliability Engineering (SRE) and private OpenShift clusters."_

> NOTE
> Red Hat recommends using PSC for more secure network connectivity.

image

Since in some cases both 'Install into an existing VPC' and 'Use Private Service Connect' are auto-selected, we should slightly modify this note. @michaelryanmcneill - what do you think? 
> IMPORTANT
> **_Private Service Connect is supported only with user created VPC_**. If you selected **Use Private Service Connect** as part of your network configuration, you must select **Install into an existing VPC**.

The above changes must also be done in [Creating a cluster on GCP with Red Hat Marketplace](https://84143--ocpdocs-pr.netlify.app/openshift-dedicated/latest/osd_install_access_delete_cluster/creating-a-gcp-cluster.html#osd-create-cluster-rhm-gcp-account_osd-creating-a-cluster-on-gcp)

The current PR is missing PSC steps in the [Creating a cluster on GCP with CCS](https://84143--ocpdocs-pr.netlify.app/openshift-dedicated/latest/osd_install_access_delete_cluster/creating-a-gcp-cluster.html#osd-create-gcp-cluster-ccs_osd-creating-a-cluster-on-gcp) section. They should be identical to changes in the GCP Marketplace and Red Hat Marketplace topics.

**Pull request merged**